app/src/main/java/com/github/campus_capture/bootcamp/map/MapScheduler.java

```diff
@@ -89,15 +92,14 @@ public void run() {
         @Override
         public void run()
         {
-            CompletableFuture<Map<String, Section>> val = CompletableFuture.supplyAsync(firebaseInterface::getCurrentZoneOwners);
-            try
-            {
-                zoneState = val.get();
-            }
-            catch(Exception e)
-            {
-                Log.e("MapScheduler", "Error ocurred when retrieving the zone owners");
-            }
+            backendInterface.getCurrentZoneOwners()
+                    .thenAccept( (result) -> {
+                        zoneState = result;
+                    }).exceptionally( e -> {
+                        // TODO handle errors better ?
```

Comment:
For now this will do the trick. The only thing which comes to mind would be to show a Toast, but that would have to be done in the fragment and not the scheduler. At least having the feedback in the log will be enough while this is still in development.